packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/example/ProtocolTestWriter.java

```diff
@@ -83,9 +79,20 @@ public ProtocolTestWriter(TestContext testContext) {
                         String defaultValueExpression = serviceClientProperty.getDefaultValueExpression();
                         String expr;
                         if (defaultValueExpression == null) {
-                            expr = String.format("Configuration.getGlobalConfiguration().get(\"%1$s\", %2$s)",
-                                serviceClientProperty.getName().toUpperCase(Locale.ROOT), ClassType.STRING
-                                    .defaultValueExpression(serviceClientProperty.getName().toLowerCase(Locale.ROOT)));
+                            if (serviceClientProperty.getType() instanceof EnumType) {
+                                expr = String.format(
+                                    "%1$s.fromString(Configuration.getGlobalConfiguration().get(\"%2$s\", %3$s))",
+                                    serviceClientProperty.getType(),
+                                    serviceClientProperty.getName().toUpperCase(Locale.ROOT),
+                                    ClassType.STRING.defaultValueExpression(
+                                        serviceClientProperty.getName().toLowerCase(Locale.ROOT)));
+
+                            } else {
+                                expr = String.format("Configuration.getGlobalConfiguration().get(\"%1$s\", %2$s)",
+                                    serviceClientProperty.getName().toUpperCase(Locale.ROOT),
+                                    ClassType.STRING.defaultValueExpression(
+                                        serviceClientProperty.getName().toLowerCase(Locale.ROOT)));
+                            }
```

Comment:
Could there be problem of non-string enum here? @XiaofeiCao 

Though I guess the probability that we have non-string enum here be pretty low. (but if it could happen, better to add a TODO for reminder)

Reply:
Yeah, for EnumType's deserialization method, we should call `((EnumType) serviceClientProperty.getType()).getFromMethodName()`.